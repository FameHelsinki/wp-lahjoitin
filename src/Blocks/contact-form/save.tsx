import { useBlockProps } from '@wordpress/block-editor'
import React from 'react'
import ContactInputGroup from './ContactInputGroup.tsx'
import ContactInputControl from './ContactInputControl.tsx'
import { SaveProps } from '../common/types.ts'

/**
 * The save function defines the way in which the different attributes should
 * be combined into the final markup, which is then serialized by the block
 * editor into `post_content`.
 *
 * @param root0
 * @param root0.attributes
 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#save
 */
export default function save({ attributes }: SaveProps): React.JSX.Element {
	const { contact, showAddress, showPhone } = attributes

	return (
		<div
			{...useBlockProps.save({ className: 'contact-form' })}
			data-contact={contact || undefined}
		>
			<ContactInputGroup.Content
				className="contact-form__row"
				name="name"
				controls={[
					{
						name: 'first_name',
						required: contact,
						type: 'text',
					},
					{
						name: 'last_name',
						required: contact,
						type: 'text',
					},
				]}
				attributes={attributes}
			/>
			<ContactInputControl.Content
				className="contact-form__row"
				name="email"
				type="email"
				required={contact}
				attributes={attributes}
			/>
			{showAddress && (
				<>
					<ContactInputControl.Content
						className="contact-form__row"
						name="address"
						type="text"
						attributes={attributes}
					/>
					<ContactInputGroup.Content
						className="contact-form__row"
						name="city_postal_code"
						controls={[
							{ name: 'city', type: 'text' },
							{ name: 'postal_code', type: 'text' },
						]}
						attributes={attributes}
					/>
				</>
			)}
			{showPhone && (
				<ContactInputControl.Content
					className="contact-form__row"
					name="phone"
					type="tel"
					attributes={attributes}
				/>
			)}
		</div>
	)
}
