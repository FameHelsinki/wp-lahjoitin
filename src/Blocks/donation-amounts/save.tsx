import { RichText, useBlockProps } from '@wordpress/block-editor'
import React from 'react'
import { SaveProps } from '../common/types.ts'
import { DEFAULT_AMOUNT, DEFAULT_LEGEND } from '../common/donation-amount.ts'
import { Attributes } from './edit.tsx'
import { __ } from '@wordpress/i18n'

type SaveAttributes = Attributes & {
	type?: boolean
}

/**
 * The save function defines the way in which the different attributes should
 * be combined into the final markup, which is then serialized by the block
 * editor into `post_content`.
 */
export default function save({ attributes }: SaveProps<SaveAttributes>): React.JSX.Element {
	const { settings, other, otherLabel, showLegend, legend } = attributes

	// Visible if other amount is shown or amount buttons is not empty.
	const visible = other || settings?.some(type => type?.amounts?.length)
	const blockProps = useBlockProps.save({
		className: visible
			? 'fame-form__fieldset fame-form__fieldset--amounts'
			: 'fame-form__hidden',
	})

	if (!visible) {
		return (
			<div {...blockProps}>
				<input
					name="amount"
					type="hidden"
					value={settings?.find(type => type.default)?.defaultAmount || DEFAULT_AMOUNT}
				/>
			</div>
		)
	}

	return (
		<fieldset {...blockProps}>
			<RichText.Content
				tagName="legend"
				className={'fame-form__legend' + (showLegend ? '' : ' screen-reader-text')}
				value={legend || DEFAULT_LEGEND}
			/>

			{settings?.map(type => (
				<div
					key={type.type}
					className={`donation-amounts donation-amounts--${type.type}`}
					data-type={type.type}
					data-default={type.default || undefined}
					data-default-amount={type.defaultAmount ?? DEFAULT_AMOUNT}
					style={{ display: type.default ? undefined : 'none' }}
				>
					{type.amounts?.map(({ value }, idx) => (
						<div className="fame-form__group" key={value}>
							<label
								htmlFor={`${type.type}-amount-${idx}`}
								className="fame-form__label"
							>
								<input
									data-type={type.type}
									className="fame-form__input"
									id={`${type.type}-amount-${idx}`}
									checked={value?.toString() === type.defaultAmount?.toString()}
									name={`amount-radio-${type.type}`}
									value={value}
									type="radio"
								/>
								{value} <span className="donation-amounts__unit">{type.unit}</span>
							</label>
						</div>
					))}
					{other && (
						<div className="donation-amounts__other">
							<RichText.Content
								htmlFor={`${type.type}-other`}
								tagName="label"
								value={otherLabel || __('Other amount', 'fame_lahjoitukset')}
							/>
							<input
								id={`${type.type}-other`}
								name={`amount-${type.type}`}
								type="number"
								value={type.defaultAmount ?? DEFAULT_AMOUNT}
								aria-describedby={`${type.type}-other-unit`}
							/>
							<span id={`${type.type}-other-unit`} className="donation-amounts__unit">
								{type.unit}
							</span>
						</div>
					)}
				</div>
			))}

			{/*
				The server only cares about `name` field. This hidden
				input field must be kept up to date with radio buttons
				and free amount field with javascript.
			*/}
			<input
				name="amount"
				type="hidden"
				value={settings?.find(type => type.default)?.defaultAmount || DEFAULT_AMOUNT}
			/>
		</fieldset>
	)
}
