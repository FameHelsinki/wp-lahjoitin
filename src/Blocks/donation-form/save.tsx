import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor'
import React from 'react'
import { SaveProps } from '../common/types.ts'

/**
 * The save function defines the way in which the different attributes should
 * be combined into the final markup, which is then serialized by the block
 * editor into `post_content`.
 *
 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#save
 */
export default function save({ attributes }: SaveProps): React.JSX.Element {
	const { returnAddress, campaign, token } = attributes

	const blockProps = useBlockProps.save({ className: 'fame-form-container' })
	const innerBlockProps = useInnerBlocksProps.save({
		className: 'fame-form__wrapper',
	})

	return (
		<div {...blockProps}>
			<form className="fame-form fame-form--donations" data-token={token || undefined} noValidate>

				<div {...innerBlockProps} />

				<input type="hidden" name="return_address" value={returnAddress || '/'} />

				{/** @todo implement configurable providers */}
				<input type="hidden" name="provider" value="mobilepay" />

				{campaign && <input type="hidden" name="campaign" value={campaign} />}
			</form>
			<div className="fame-form-overlay">
				<div className="fame-form-spinner" role="status">
					<span className="screen-reader-text">Loading</span>
				</div>
			</div>
		</div>
	)
}
