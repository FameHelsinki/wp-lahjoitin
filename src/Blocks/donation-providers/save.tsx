import React from 'react'
import { useBlockProps } from '@wordpress/block-editor'
import { SaveProps } from '../common/types.ts'

type Provider = { value: string; label: string }

export default function Save({
	attributes,
}: SaveProps<{
	providers?: Record<string, Provider[]>
	legend?: string
	showLegend?: boolean
}>): React.JSX.Element {
	const { providers = {}, legend = 'Provider type', showLegend = true } = attributes
	const flatProviders = Array.isArray(providers) ? providers : []

	return (
		<div {...useBlockProps.save()}>
			<fieldset className="payment-method-selector">
				{showLegend && <legend className="fame-form__legend">{legend}</legend>}

				{flatProviders.map(p => (
					<div key={`${p.type}-${p.value}`} data-type={p.type}>
						<label htmlFor={`payment_method_${p.type}_${p.value}`}>
							<input
								type="radio"
								id={`payment_method_${p.type}_${p.value}`}
								name="provider"
								value={p.value}
								data-type={p.type}
								required
							/>
							<span className="provider-type__label">{p.label}</span>
						</label>
					</div>
				))}
			</fieldset>
		</div>
	)
}
