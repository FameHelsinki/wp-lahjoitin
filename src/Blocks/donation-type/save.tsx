import { RichText, useBlockProps } from '@wordpress/block-editor'
import React from 'react'
import { DEFAULT_DONATION_TYPE } from '../common/donation-type.ts'
import { SaveProps } from '../common/types.ts'
import { Attributes } from './edit.tsx'

/**
 * The save function defines the way in which the different attributes should
 * be combined into the final markup, which is then serialized by the block
 * editor into `post_content`.
 *
 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#save
 */
export default function save({ attributes }: SaveProps<Attributes>): React.JSX.Element {
	const { legend, showLegend, types, value: defaultValue } = attributes

	if (!Array.isArray(types) || types.length <= 1) {
		return (
			<input
				{...useBlockProps.save({ className: 'donation-type' })}
				type="hidden"
				name="type"
				value={types?.[0]?.value ?? DEFAULT_DONATION_TYPE.value}
			/>
		)
	}

	return (
		<fieldset {...useBlockProps.save({ className: 'donation-type' })}>
			<RichText.Content
				tagName="legend"
				className={'donation-type__legend' + (showLegend ? '' : ' screen-reader-text')}
				value={legend ?? ''}
			/>
			<div className="donation-type__controls">
				{types.map(({ value, label }) => (
					<label
						key={value}
						htmlFor={`donation-type-${value}`}
						className="donation-type__label"
					>
						<input
							id={`donation-type-${value}`}
							className="donation-type__input"
							checked={value === defaultValue}
							type="radio"
							name="type"
							value={value}
						/>
						{label}
					</label>
				))}
			</div>
		</fieldset>
	)
}
