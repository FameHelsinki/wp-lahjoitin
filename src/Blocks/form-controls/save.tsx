import { __ } from '@wordpress/i18n'
import { RichText, useBlockProps } from '@wordpress/block-editor'
import React from 'react'
import { SaveProps } from '../common/types.ts'

/**
 * The save function defines the way in which the different attributes should
 * be combined into the final markup, which is then serialized by the block
 * editor into `post_content`.
 *
 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#save
 */
const Save = ({ attributes: { submitLabel } }: SaveProps): React.JSX.Element => {
	// Submit button has disabled attribute by default.
	// The attribute is removed when JavaScript is loaded.
	return (
		<div {...useBlockProps.save({ className: 'fame-form__controls' })}>
			<RichText.Content
				disabled
				type="submit"
				className="wp-element-button is-primary"
				tagName="button"
				value={submitLabel}
			/>
			<noscript>
				{__('Please enable JavaScript to use this form.', 'fame_lahjoitukset')}
			</noscript>
		</div>
	)
}

export default Save
