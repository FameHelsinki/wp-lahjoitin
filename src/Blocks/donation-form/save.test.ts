import { describe, it, expect } from '@jest/globals'
import { render } from '@testing-library/react'
import save from './save'

// Mock attributes for testing
const mockAttributes = {
	returnAddress: 'https://example.com/return',
	campaign: 'test-campaign',
	token: true,
}

describe('Gutenberg Block Save Function', () => {
	it('renders correctly and matches snapshot', () => {
		const { container } = render(save({ attributes: mockAttributes }))
		expect(container).toMatchSnapshot()
	})
})
