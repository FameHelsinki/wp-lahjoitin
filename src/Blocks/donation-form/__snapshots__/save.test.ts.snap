// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gutenberg Block Save Function renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="mock-block-props"
  >
    <form
      class="fame-form fame-form--donations"
      data-token="true"
      novalidate=""
    >
      <div
        class="mock-inner-block-props"
      />
      <input
        name="return_address"
        type="hidden"
        value="https://example.com/return"
      />
      <input
        name="provider"
        type="hidden"
        value="mobilepay"
      />
      <input
        name="campaign"
        type="hidden"
        value="test-campaign"
      />
    </form>
    <div
      class="fame-form-overlay"
    >
      <div
        class="fame-form-spinner"
        role="status"
      >
        <span
          class="screen-reader-text"
        >
          Loading
        </span>
      </div>
    </div>
  </div>
</div>
`;
